troposphere/rds.py

```diff
@@ -187,6 +187,13 @@ def validate(self):
                 'AWS::RDS::DBInstance.'
             )
 
+        if 'Engine' not in self.properties and \
+           'DBSnapshotIdentifier' not in self.properties:
+            raise ValueError(
+                'Either Engine or DBSnapshotIdentifier is required '
+                'AWS::RDS::DBInstance.'
+            )
+
```

Comment:
Since Engine can be optional when specifying DBSnapshotIdentifier, perhaps this could be coded:
```
        if 'DBSnapshotIdentifier' not in self.properties:
            if 'Engine' not in self.properties:
                raise ValueError(
                    'Resource Engine required in type %s' % self.resource_type)```